platform/public_api_guide.md

```diff
@@ -0,0 +1,59 @@
+Public API guide
+========================
+
+This page describes how to access Piwik PRO API which uses 
+[client credentials](https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/)
+Oauth grant type for obtaining user token. 
+All data is sent and received as JSON and it uses [JSON API](http://jsonapi.org/) specification.
+
+## Obtaining token
+
+If you want to access API for the first time you need to generate your 
+API Credentials which then allows you to request for a token that is used for authentication during communication with public API.
+
+### Generate API Credentials
+
+* Login to your account using your email and password.
+* Go to your profile (`Menu` then `My profile`).
+* On this page click on `API Credentials` tab. This page allows you to manage all your API credentials.
+* Click `Generate new credentials` which will result in new popup. Fill in your custom `credentials name`.
+  Name must contains at least 3 characters.
+* Copy your newly generated `CLIENT ID` and `CLIENT SECRET` because they **won't be available for you after dismissing this window.**
+
+### Create access token
+
+Having generated your API Credentials, now you are ready for creating access token that will be used in communication with API.
+To do so, send POST request 
+```
+POST /auth/token
+```
+
+Curl example:
+
+```
+curl 'https://<domain>/auth/token' -H "Content-Type: application/json" --data '{
```

Comment:
Is this endpoint also documented in OpenAPI?

Reply:
Can't find it there.

Reply:
no, this endpoint is not documented via OpenAPI. We wanted to describe it via public api guide because it's some kind of special and not JsonApi complaint. Also we have other endpoints in `OAuth` service, but docs are unfortunately not ready yet.

Reply:
Users will assume that every endpoint is described to together. Splitting endpoint description like that may not be worth it if it'll cause confusion.

Reply:
Our whole documentation lacks information about which endpoints are protected with OAuth/JWT (tag `security`) and which aren't so at the moment user is still obliged to pass through this section if he wants to use our API.

Im not arguing that this will not be better but I think that we could fix this in next iteration of this docs when we have some feedback from end users.